packages/app/src/sections/dashboard/SpotHistoryTable.tsx

```diff
@@ -27,34 +26,39 @@ const conditionalRender = <T,>(prop: T, children: ReactElement) =>
 
 const SpotHistoryTable: FC = () => {
 	const { t } = useTranslation()
-	const { network, walletAddress } = Connector.useContainer()
+	const { network } = Connector.useContainer()
 	const synthsMap = useAppSelector(selectSynthsMap)
 	const { selectPriceCurrencyRate, selectedPriceCurrency } = useSelectedPriceCurrency()
-	const walletTradesQuery = useGetWalletTrades(walletAddress!)
 
 	const synths = useMemo(() => Object.values(synthsMap) || [], [synthsMap])
-	const trades = useMemo(() => {
-		const t = walletTradesQuery.data?.synthExchanges ?? []
 
-		return t.map((trade: any) => ({ ...trade, hash: trade.id.split('-')[0] }))
-	}, [walletTradesQuery.data])
+	const dispatch = useAppDispatch()
+
+	useEffect(() => {
+		dispatch(fetchWalletTrades())
+	}, [dispatch])
+
+	const walletTrades = useAppSelector(selectWalletTrades)
```

Comment:
Be nice to keep the useAppSelector calls together just to keep things tidy

Reply:
Deleted table.